pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -958,18 +975,26 @@ cdef _timedelta_from_value_and_reso(cls, int64_t value, NPY_DATETIMEUNIT reso):
         _Timedelta td_base
 
     assert value != NPY_NAT
-    # For millisecond and second resos, we cannot actually pass int(value) because
-    #  many cases would fall outside of the pytimedelta implementation bounds.
-    #  We pass 0 instead, and override seconds, microseconds, days.
-    #  In principle we could pass 0 for ns and us too.
+    # In order to not break the C-API for pytimedeltas,
+    # we pass values when they are within bounds
+    # and pass zero when values are not within bounds.
+    # Thus pytimedelta C-API values are stop working only when they
+    # should, while allowing Pandas Timedeltas to work for
+    # a broader set of values than the pytimedelta bounds.
     if reso == NPY_FR_ns:
         td_base = _Timedelta.__new__(cls, microseconds=int(value) // 1000)
     elif reso == NPY_DATETIMEUNIT.NPY_FR_us:
         td_base = _Timedelta.__new__(cls, microseconds=int(value))
     elif reso == NPY_DATETIMEUNIT.NPY_FR_ms:
-        td_base = _Timedelta.__new__(cls, milliseconds=0)
+        if value < PYTIMEDELTA_UPPER_MS and value >= PYTIMEDELTA_LOWER_MS:
+            td_base = _Timedelta.__new__(cls, milliseconds=int(value))
+        else:
+            td_base = _Timedelta.__new__(cls, milliseconds=0)
```

Comment:
Wouldn't we just want to raise before sending to the C-API for pytimedeltas? Using 0 here is a bit arbitrary

Reply:
Good question. I left it as zero thinking that might help with backwards-compatibility, since that was the previous behavior. I think (though I may be wrong, please correct me) that pandas supports a wider range of timedelta values than python, so we don't want to break existing code that leverages that feature of pandas in an attempt to make these python C-APIs work.

Perhaps we raise a warning rather than an error? Quietly sending a zero was part of the problem with the previous implementation, so perhaps we should at least raise a warning showing that the zero behavior is intentional.

Reply:
I coded up a few possibilities with a warning, then ended up scratching them all due to the potential confusion to users. So how about this simple implementation? Or if this sparks any different ideas for you, I'm all ears!

Reply:
Hmm I still don't really understand the point of setting to 0 - wouldn't we want to just let these raise? If you do so you can really simplify your test cases as well; just set up scenarios that we know should and shouldn't exceed the bounds and test the constructor against those

@jorisvandenbossche for thoughts in case I am overlooking something

Reply:
I'm hesitant to let them raise given the original comment in the code, which reads:

```
# For millisecond and second resos, we cannot actually pass int(value) because 
#  many cases would fall outside of the pytimedelta implementation bounds. 
```
 
@jorisvandenbossche also alluded to cases in which pd.Timedelta allows for values outside of the range allowed by pytimedelta, but I've been unable to verify if any of these actually exist.

Agree that we need further clarity -- if there's no supported case in pandas in which a pytimedelta error would be raised, then we should just let the pytimedelta errors raise. But the comment in the code and Joris' original description make me concerned that there might be such cases, in which case we want to allow Pandas to proceed on without pytimedelta raising an error.

Reply:
>  I think (though I may be wrong, please correct me) that pandas supports a wider range of timedelta values than python

Correct.  Non-nanosecond support for timestamps/timedeltas means we can create Timestamps/Timedeltas that the stdlib datetime/timedelta cannot handle.  Passing zero here was the best idea I had, but I'm open to other options.  I don't think "just raise" is compatible with current non-nano support.

Reply:
Our timedelta still inherits from the CPython timedelta right? If so, I _think_ we need to first figure out a way to decouple that if we plan on supporting things outside of the range as a precursor. My main concern is that we'd be fighting the inheritance tree otherwise

Reply:
Timedelta subclasses pytimedelta yes.  But the C-API does not respect subclasses, so we aren't doing anything wrong inheritance tree-wise